cdap-docs/examples-manual/source/examples/stream-conversion.rst

```diff
@@ -0,0 +1,163 @@
+  the output, which is in Avro format::
+
+    Job job = context.getHadoopJob();
+    job.setMapperClass(StreamConversionMapper.class);
+    job.setNumReduceTasks(0);
+    job.setMapOutputKeyClass(AvroKey.class);
+    job.setMapOutputValueClass(NullWritable.class);
+    AvroJob.setOutputKeySchema(job, SCHEMA);
+
+- Based on the logical start time, determine the range of events to read from the stream::
+
+    // read 5 minutes of events from the stream, ending at the logical start time of this run
+    long logicalTime = context.getLogicalStartTime();
+    StreamBatchReadable.useStreamInput(context, "events", logicalTime - TimeUnit.MINUTES.toMillis(5), logicalTime);
+
+- Each run writes its output to a partition with the logical start time::
+
+    TimePartitionedFileSetArguments.setOutputPartitionTime(dsArguments, logicalTime);
+    TimePartitionedFileSet partitionedFileSet = context.getDataset("converted", dsArguments);
+    context.setOutput("converted", partitionedFileSet);
+
+- Note that the output file path is derived from the output partition time by the dataset itself::
+
+    LOG.info("Output location for new partition is: {}",
+             partitionedFileSet.getUnderlyingFileSet().getOutputLocation().toURI().toString());
+
+The Mapper itself is straight-forward: For each event, it emits an Avro record:
+
+.. literalinclude:: /../../../cdap-examples/StreamConversion/src/main/java/co/cask/cdap/examples/streamconversion/StreamConversionMapReduce.java
+     :language: java
+     :lines: 104-111
+
+In the ``afterSubmit`` method of the MapReduce program, if the run succeeded, the output file is
+registered as a new partition in the ``converted`` dataset. Note that this is due to a gap in the
+implementation of ``TimePartitionedFileSet``, which should add the partition in the output committer
+of its output format. This will be addressed in future CDAP releases:
+
+.. literalinclude:: /../../../cdap-examples/StreamConversion/src/main/java/co/cask/cdap/examples/streamconversion/StreamConversionMapReduce.java
+     :language: java
+     :lines: 85-95
+
+Building and Starting
+=====================
+
+- You can either build the example (as described `below
+  <#building-an-example-application>`__) or use the pre-built JAR file included in the CDAP SDK.
+- Start CDAP and deploy the application as described below in `Running CDAP Applications`_\ .
+- Once the application has been deployed and started, you can `run the example. <#running-the-example>`__
+
+Running CDAP Applications
+=========================
+
+.. include:: /../../developers-manual/source/getting-started/building-apps.rst
+    :start-line: 9
+
+Running the Example
+===================
+
+The ``StreamConversionWorkflow`` will run automatically every 5 minutes based on its schedule.
+To give it some data, you can use the provided script to send events to the stream, for example,
```

Comment:
Looks like I'm the one getting tired now. :)

use a provided script to send events to the stream; for example,